### PR TITLE
Fix Environment retrieve methods

### PR DESCRIPTION
**Pull request by DimaMishchenko**
### Description 📝
Fix of `Environment` variable retrieve methods provided in #1468

Currently this functionality works incorrect. 
If variable is not exists, `dynamicMember` will return `nil` and `getString`/`getBoolean` will never be called.
Also in case if variable exist and you use helper methods you still will get optional value.
Can be fixed by creating extension to `Optional` type.

### Example 
``` swift
// appName not exists

// before
Environment.appName?.getString(default: "TuistApp") // nil

// now
Environment.appName.getString(default: "TuistApp") // "TuistApp"
```

### Checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase.
- [x] The changes have been tested following the [documented guidelines](https://tuist.io/docs/contribution/testing-strategy/).
- [x] The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.
- [x] In case the PR introduces changes that affect users, the documentation has been updated.

### CODE REVIEW

File: Sources/ProjectDescription/Environment.swift [hunk 1]
```diff
@@ -5,24 +5,6 @@ public struct Environment {
     public enum Value {
         case boolean(Bool)
         case string(String)
-
-        /// Retrieve the Environment value as a string or return the specified default string value
-        /// - Parameters:
-        ///   - default: default String value to be returned
-        /// - Returns: String
-        public func getString(default defaultString: String) -> String {
```
Comment (fortmarek) [inline Sources/ProjectDescription/Environment.swift:13]: can we just mark this `@deprecated`? Otherwise, this change would be breaking which is something we should try to avoid.

**Review by fortmarek — COMMENTED** [review]
Hi @DimaMishchenko! Changes look good, as mentioned, it'd be great to make this change non-breaking

File: Sources/ProjectDescription/Environment.swift (hunk 1 above)
Comment (DimaMishchenko) [reply Sources/ProjectDescription/Environment.swift:13]: Makes sense. Will do.

File: Sources/ProjectDescription/Environment.swift [hunk 2]
```diff
@@ -10,6 +10,7 @@ public struct Environment {
         /// - Parameters:
         ///   - default: default String value to be returned
         /// - Returns: String
+        @available(*, deprecated, message: "Was moved to Optional extension. Remove question mark to fix.")
```
Comment (fortmarek) [inline Sources/ProjectDescription/Environment.swift:13]: I'd rephrase it, user does not have to really know the implementation details - something like: `Environment variables as optionals are deprecated. Remove question mark to fix deprecation warning.`

**Review by fortmarek — APPROVED** [review]
Looks good to me - I'd just change the deprecation message but otherwise 👍 from me

**Review by kwridan — APPROVED** [review]
Good catch @DimaMishchenko - thanks for fixing this 👍 

One additional suggestion would be to perhaps update the fixture https://github.com/tuist/tuist/blob/main/fixtures/framework_with_environment_variables/Project.swift to leverage the new syntax.

**Comment by DimaMishchenko** [comment]
Done. Thank you @fortmarek and @kwridan

**Comment by fortmarek** [comment]
let's merge this once CI is green, unfortunately, we have some issues on `main`. We should re-run the tests here once that is fixed.

**Comment by pepibumur** [comment]
@all-contributors add @DimaMishchenko for code

**Comment by fortmarek** [comment]
@DimaMishchenko could you please rebase and see if the pipeline succeeds now? The `main` should be fixed now. Sorry for the additional troubles 🙂
